src/main/scala/com/actian/spark_vector/vector/Vector.scala

```diff
@@ -53,8 +53,9 @@ private[vector] object Vector extends Logging {
     } else {
       (rdd, rddSchema)
     }
-    fillWithNulls(inputRDD, inputType, targetSchema,
-      field2Columns.map(i => i.columnName -> i.fieldName).toMap)
+    val colMappingOpt = targetToInput(inputType, targetSchema, field2Columns.map(i => i.columnName -> i.fieldName).toMap)
+    logDebug(s"Mapping of cols before load is $colMappingOpt for inputTypeFields = ${inputType.fields.map(_.name).mkString(",")}, targetTypeFields = ${targetSchema.fields.map(_.name).mkString(",")}")
+    if (colMappingOpt.isDefined) rdd.map(row => colMappingOpt.get.map { i => if (i.isDefined) row(i.get) else null }) else rdd
```

Comment:
It's not clear to me what should happen when you have a column that's not found in the mapping.. maybe we should just error out instead of attempting to insert null